IDEMSInternational/parenting-app-ui
# Refactor: Changed oauth method used for authenticating author's google account to access gdrive

PR Checklist


## Description

In order for an author/dev to sync data from Google Drive, they are required to authenticate using their Google account. The method we previously used for this is being deprecated by Google on October 3rd (details [here](https://developers.google.com/identity/protocols/oauth2/resources/oob-migration)). This PR implements a new oauth flow (namely the [Loopback IP Address flow](https://developers.google.com/identity/protocols/oauth2/native-app#redirect-uri_loopback)) in order to generate and receive the necessary auth token.

Also updates readme to replace references to deprecated scripts.

## Testing

In order to test the auth flow:
1. Checkout this branch, `refactor/google-oauth-flow`
2. Run `yarn scripts config decrypt` to repopulate `packages/scripts/config/credentials.json` with updated credentials
3. Delete the auth token saved to `packages/scripts/config/token.json`
4. Run `yarn scripts app-data download --authorize` and complete the new authorisation flow
5. A new token should be saved to `packages/scripts/config/token.json`
6. Run `yarn workflow sync` and pay attention to the output to ensure that the authorisation has worked correctly and you are able to download from Google Drive (e.g. make a trivial change to a spreadsheet to ensure it gets synced correctly).

## Git Issues

Closes #1517 

## Screenshots/Videos

A video showing completion of the new authorisation flow:


https://user-images.githubusercontent.com/64838927/192834787-1fd90ae6-20f5-4aa7-b091-064526aa9c14.mov
